File: backups/rag_data_restore_20260607_013709/frontend_src/api/researchExport.ts
import { apiClient } from "./client";

export type ResearchSummary = {
  total_participants: number;
  risk_distribution: Record<string, number>;
  cluster_distribution: Record<string, number>;
  cluster_risk_overlay?: Record<string, Record<string, number>>;
  prescription_status: Record<string, number>;
  template_effects?: Record<string, number>;
  export_job_status?: Record<string, number>;
  intervention_effects: {
    feedback_count: number;
    average_completion_rate: number;
    average_rpe: number;
    discomfort_event_count: number;
    pain_worsened_count: number;
    completion_rate_trend?: Array<{ date: string; value: number }>;
    rpe_trend?: Array<{ date: string; value: number }>;
    pain_trend?: Array<{ date: string; value: number }>;
    blood_pressure_trend?: Array<{ date: string; sbp?: number; dbp?: number }>;
    blood_glucose_trend?: Array<{ date: string; value: number }>;
  };
};

export type DesensitizedUserRow = {
  research_subject_id?: string;
  participant_code: string;
  profile: Record<string, unknown>;
  fitness_test: Record<string, unknown> | null;
  risk_screening: Record<string, unknown> | null;
  latest_prescription: Record<string, unknown> | null;
};

export type ResearchExportResponse = {
  items: DesensitizedUserRow[];
  total: number;
};

export type ResearchExportFormat = "csv" | "xlsx" | "json";

export type ResearchExportRequest = {
  id: number;
  requested_by: number;
  organization_id?: number | null;
  format: ResearchExportFormat;
  purpose: string;
  status: string;
  approved_by: number | null;
  approval_comment: string | null;
  row_count: number;
  expires_at: string | null;
  downloaded_at: string | null;
  created_at: string;
};

export function getResearchSummary(): Promise<ResearchSummary> {
  return apiClient.get("/research/export/summary").then((response) => response.data);
}

export function exportDesensitizedUsers(): Promise<ResearchExportResponse> {
  return apiClient.get("/research/export/users").then((response) => response.data);
}

export function listResearchExportRequests(): Promise<ResearchExportRequest[]> {
  return apiClient.get("/research/export/requests").then((response) => response.data);
}

export function createResearchExportRequest(payload: { format: ResearchExportFormat; purpose: string }): Promise<ResearchExportRequest> {
  return apiClient.post("/research/export/requests", payload).then((response) => response.data);
}

export function approveResearchExportRequest(
  requestId: number,
  payload: { approval_comment: string }
): Promise<ResearchExportRequest> {
  return apiClient.post(`/research/export/requests/${requestId}/approve`, payload).then((response) => response.data);
}

export function rejectResearchExportRequest(
  requestId: number,
  payload: { approval_comment: string }
): Promise<ResearchExportRequest> {
  return apiClient.post(`/research/export/requests/${requestId}/reject`, payload).then((response) => response.data);
}

export function downloadResearchExportRequest(requestId: number): Promise<Blob> {
  return apiClient.get(`/research/export/requests/${requestId}/download`, { responseType: "blob" }).then((response) => response.data);
}

File: backups/prod_deploy_ui_polish_20260607_021154/backend_app/schemas/research_export.py
from typing import Any

from pydantic import BaseModel, Field


class DesensitizedUserRow(BaseModel):
    participant_code: str
    profile: dict[str, Any]
    fitness_test: dict[str, Any] | None
    risk_screening: dict[str, Any] | None
    latest_prescription: dict[str, Any] | None


class ResearchExportResponse(BaseModel):
    items: list[DesensitizedUserRow]
    total: int


class TrendPoint(BaseModel):
    date: str
    value: float


class BloodPressureTrendPoint(BaseModel):
    date: str
    sbp: float | None = None
    dbp: float | None = None


class InterventionEffects(BaseModel):
    feedback_count: int
    average_completion_rate: float
    average_rpe: float
    discomfort_event_count: int
    pain_worsened_count: int
    completion_rate_trend: list[TrendPoint] = Field(default_factory=list)
    rpe_trend: list[TrendPoint] = Field(default_factory=list)
    pain_trend: list[TrendPoint] = Field(default_factory=list)
    blood_pressure_trend: list[BloodPressureTrendPoint] = Field(default_factory=list)
    blood_glucose_trend: list[TrendPoint] = Field(default_factory=list)


class ResearchSummaryResponse(BaseModel):
    total_participants: int
    risk_distribution: dict[str, int]
    cluster_distribution: dict[str, int]
    cluster_risk_overlay: dict[str, dict[str, int]] = Field(default_factory=dict)
    prescription_status: dict[str, int]
    template_effects: dict[str, float] = Field(default_factory=dict)
    export_job_status: dict[str, int] = Field(default_factory=dict)
    intervention_effects: InterventionEffects


class ResearchExportRequestCreate(BaseModel):
    format: str = Field(pattern="^(csv|xlsx|json)$")
    purpose: str = Field(min_length=2, max_length=500)


class ResearchExportApproval(BaseModel):
    approval_comment: str = Field(min_length=1, max_length=500)


class ResearchExportRequestRead(BaseModel):
    id: int
    requested_by: int
    organization_id: int | None = None
    format: str
    purpose: str
    status: str
    approved_by: int | None
    approval_comment: str | None
    row_count: int
    expires_at: str | None
    downloaded_at: str | None
    created_at: str
